### DISCUSSION TITLE
Could not resolve path '_id' against

### DISCUSSION THREAD

**Discussion by zjie1595** [MorphiaOrg/morphia#2109] [General]
** Please ask your question and provide whatever background or supporting information you think might be relevant: **

i try to save this class
<img width="738" alt="Snipaste_2022-10-09_09-45-31" src="https://user-images.githubusercontent.com/66541098/194733727-c6ad88a8-46c3-4420-9a52-5e6ae5af772b.png">
<img width="476" alt="Snipaste_2022-10-09_09-45-59" src="https://user-images.githubusercontent.com/66541098/194733733-25586314-7c97-4fbb-9caa-0b0d99949fee.png">
but failed
<img width="1748" alt="Snipaste_2022-10-09_09-48-40" src="https://user-images.githubusercontent.com/66541098/194733776-c187266c-1ff8-491b-9c06-3a83820e97bf.png">
please tell me how to save this class,or any example?
** Please complete the following information: **
    Server Version:
    Driver Version:
    Morphia Version:2.2.9

**Comment by evanchooly** [comment]
You need to have a field with `@Id` on it in your referenced entity.

**Reply by evanchooly** [reply]
You will also need to save those referenced entities first.  when you try to do that, as things are now, Morphia should complain about that missing `@Id`.

**Reply by zjie1595** [reply]
episode is not top document,how can i handle this type?

**Reply by zjie1595** [reply]
if i use morphiaRefrence,it worked,but i want to know how to work with @Reference

**Reply by evanchooly** [reply]
Is Episode going to live in its own collection?  Then it needs an ID field.  If not, don't use @Reference on that Map.

**Reply by zjie1595** [reply]
thanks
